File: generated-code/codecraft/python/FileReadWrite/model/entity_type.py
from enum import IntEnum

class EntityType(IntEnum):
    """Entity type"""

    WALL = 0
    """Wall, can be used to prevent enemy from moving through"""
    HOUSE = 1
    """House, used to increase population"""
    BUILDER_BASE = 2
    """Base for recruiting new builder units"""
    BUILDER_UNIT = 3
    """Builder unit can build buildings"""
    MELEE_BASE = 4
    """Base for recruiting new melee units"""
    MELEE_UNIT = 5
    """Melee unit"""
    RANGED_BASE = 6
    """Base for recruiting new ranged units"""
    RANGED_UNIT = 7
    """Ranged unit"""
    RESOURCE = 8
    """Resource can be harvested"""
    TURRET = 9
    """Ranged attacking building"""

    def __repr__(self):
        return str(self)

File: generated-code/pascal/codecraft/FileReadWrite/Model/EntityType.pas
unit EntityType;

interface

uses
    Stream,
    SysUtils;

type
    // Entity type
    {$scopedEnums on}
    TEntityType = (
        // Wall, can be used to prevent enemy from moving through
        Wall = 0,
        // House, used to increase population
        House = 1,
        // Base for recruiting new builder units
        BuilderBase = 2,
        // Builder unit can build buildings
        BuilderUnit = 3,
        // Base for recruiting new melee units
        MeleeBase = 4,
        // Melee unit
        MeleeUnit = 5,
        // Base for recruiting new ranged units
        RangedBase = 6,
        // Ranged unit
        RangedUnit = 7,
        // Resource can be harvested
        Resource = 8,
        // Ranged attacking building
        Turret = 9);

implementation

end.